src/dune_data_loading.rs

```diff
@@ -0,0 +1,86 @@
+extern crate serde_derive;
+use anyhow::Result;
+use primitive_types::H160;
+use serde_json;
+use std::collections::HashMap;
+use std::fs::File;
+use std::io::BufReader;
+use std::path::Path;
+
+use crate::models::dune_json_formats::{Data, DuneJson};
+use crate::models::in_memory_database::InMemoryDatabase;
+
+pub fn load_dune_data_into_memory<P: AsRef<Path>>(path: P) -> Result<InMemoryDatabase> {
+    let dune_json = read_dune_data_from_file(path)?;
+    load_data_from_json_into_memory(dune_json)
+}
+
+fn read_dune_data_from_file<P: AsRef<Path>>(path: P) -> Result<DuneJson> {
+    // Open the file in read-only mode with buffer.
+    let file = File::open(path)?;
+    let reader = BufReader::new(file);
+    let u = serde_json::from_reader(reader)?;
+    Ok(u)
+}
+
+fn load_data_from_json_into_memory(dune_download: DuneJson) -> Result<InMemoryDatabase> {
```

Comment:
Nice! I like the strategy of splitting this into a separate function so the logic here can be tested without needing some test data in a separate file in the repo.